polus-vector-label-plugin/src/main.py

```diff
@@ -0,0 +1,153 @@
+    parser.add_argument('--stitch_threshold',required=False, type=float,
+                        help='Stitch threshold for 3D', default=0.0)
+    # Output arguments
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()
+    inpDir = args.inpDir
+    logger.info('inpDir = {}'.format(inpDir))
+    outDir = args.outDir
+    logger.info('outDir = {}'.format(outDir))
+    cellprob_threshold = args.cellprob_threshold
+    flow_threshold= args.flow_threshold
+    stitch_threshold=args.stitch_threshold
+    rescale = np.ones(1)
+    niter = 1 / rescale[0] * 200
+    # Surround with try/finally for proper error catching
+    try:
+        logger.info('Initializing ...')
+        # Get all file names in inpDir image collection
+        root = zarr.open(str(Path(inpDir).joinpath('flow.zarr')),mode='r')
+        count=0
+        # Loop through files in inpDir image collection and process
+        for m,l in root.groups():
+            logger.info('Processing image ({}/{}): {}'.format(count + 1, len([m for m, l in root.groups()]), m))
+            y=l['vector']
+            y= np.asarray(y)
+            metadata=l.attrs['metadata']
+
+            if len(y.shape)==4:
+                mask_stack=[]
+                for i in range(int(y.shape[0])):
+                    prob=y[i,:,:,:].astype(np.float32)
+                    cellprob = prob[:, :, -1]
+                    dP = np.stack((prob[..., 0], prob[..., 1]), axis=0)
+                    niter = 1 / rescale[0] * 200
+                    p = mask.follow_flows(-1 * dP * (cellprob > cellprob_threshold) / 5.)
+                    masks = mask.compute_masks(p, cellprob, dP, cellprob_threshold, flow_threshold)
+                    mask_stack.append(masks)
+
+                if stitch_threshold > 0.0:
+                    mask_stack = stitch3D(np.array(mask_stack), stitch_threshold=stitch_threshold)
+                maski = np.transpose(np.array(mask_stack) ,(1,2,0))
+                x,y,z = maski.shape
+                maski = np.reshape(maski,(x,y,z,1,1))
+
+            elif len(y.shape) == 3 :
+                prob=y.astype(np.float32)
+                cellprob = prob[:, :, -1]
+                dP = np.stack((prob[..., 0], prob[..., 1]), axis=0)
+                p=mask.follow_flows(-1 * dP * (cellprob > cellprob_threshold) / 5.,
+                                                                 niter=niter, interp=True)
+                maski = mask.compute_masks(p,cellprob,dP,cellprob_threshold,flow_threshold)
+                x_shape,y_shape = maski.shape
+                maski = np.reshape(maski, (x_shape,y_shape, 1, 1,1))
+
+            # Write the output
+            temp = re.sub(r'(?<=Type=")([^">>]+)', str(maski.dtype), metadata)
+            xml_metadata = OmeXml.OMEXML(temp)
+            path=Path(outDir).joinpath(str(m))
```

Comment:
Try to use more descriptive variable names. Instead of `m` as a variable, consider calling it `filename` or something that gives a better indication of what it represents. In addition to making functional code, we also want it to be readable. Please go over your code and rename variables where it makes sense so that it is easier to follow.